debugger/src/debugger.rs

```diff
@@ -592,7 +592,54 @@ impl Debugger {
                                 return false;
                             }
                         };
+                        // For some operations, we need to make sure the core isn't sleeping, by calling `Core::halt()`
+                        // When we do this, we need to flag it (`unhalt_me = true`), and later call `Core::run()` again.
+                        // NOTE: the target will exit sleep mode as a result of this command.
+                        let mut unhalt_me = false;
                         match valid_command.function_name {
+                            "set_breakpoint" | "set_breakpoints" | "clear_breakpoint"
+                            | "stack_trace" | "threads" | "scopes" | "variables"
+                            | "read_memory" | "write" | "source" => {
+                                match core_data.target_core.status() {
+                                    Ok(current_status) => {
+                                        if current_status == CoreStatus::Sleeping {
+                                            match core_data
+                                                .target_core
+                                                .halt(Duration::from_millis(100))
+                                            {
+                                                Ok(_) => {
+                                                    unhalt_me = {
```

Comment:
This extra scope can be removed.

Reply:
Done. It wasn't really extra scope, just a short-hand / style preference for chaining statements. Old habits die hard :)